src/firmware_new/tools/send_proto_over_udp.cpp

```diff
@@ -33,22 +31,18 @@ int main(int argc, char* argv[])
     control_req.mutable_wheel_2_control()->CopyFrom(WheelControl);
     control_req.mutable_wheel_2_control()->CopyFrom(WheelControl);
 
-    int count = 0;
+    // create an io service and run it in a thread to handle async calls
+    boost::asio::io_service io_service;
+    auto io_service_thread = std::thread([&]() { io_service.run(); });
 
-    // create a RobotCommunicator with a NetworkMedium
-    RobotCommunicator<ControlMsg, RobotAck> communicator(
-        std::make_unique<NetworkMedium>(std::string(AI_MULTICAST_ADDRESS) + "%eth0",
-                                        AI_MULTICAST_SEND_PORT, AI_UNICAST_LISTEN_PORT),
-        [&](const ControlMsg& msg) {
-            std::cout << "COMP Txed " << count++ << " msgs " << std::endl;
-        },
-        [&](const RobotAck& msg) {
-            std::cout << "STM32 Rxed " << msg.msg_count() << " msgs " << std::endl;
-        });
+    // create ProtoMulticastSender to send proto
+    auto sender = std::make_unique<ProtoMulticastSender<ControlMsg>>(
+        io_service, std::string(AI_MULTICAST_ADDRESS) + "%enp3s0f1",
```

Comment:
what's this: `%enp3s0f1`?

Reply:
Its the interface on my computer for ethernet, it will be slightly different for everyone so just made it an arg you pass in and added some docs on how to use it ontop.